app/models/candidate_interface/work_experience_form.rb

```diff
@@ -14,7 +14,9 @@ class WorkExperienceForm
     validates :working_with_children, inclusion: { in: %w(true false) }
 
     validate :start_date_valid
-    validate :start_date_after_end_date
+    validate :end_date_valid, if: :end_date_given?
+    validate :end_date_year_before_current_year, if: :end_date_valid?
+    validate :start_date_after_end_date, if: :start_date_and_end_date_valid?
```

Comment:
The current list of date validations:

```
start_date_valid
end_date_valid
end_date_given?
end_date_valid?
end_date_year_before_current_year
start_date_after_end_end_date
```

A couple of observations

- It's odd to me that the first two are named for the valid state and the second two for the invalid state
- We have to have the two `?` functions to make this declarative logic work
- To follow the logic we have to jump around the file quite a lot

Given that these dates are either valid together or not, have you considered having a single function which expressed the whole validation? `validate :start_and_end_dates_valid`? This function could figure out the error state and then attach errors in the appropriate places in one pass at the end.

Reply:
I've renamed `start_date_after_end_date` to `start_date_before_end_date` so that they all reflect the valid state now

Reply:
Also found & fixed a few more edge cases:

partially-blank end date
end date year == current year, but month > current month

@duncanjbrown how does this look now?